app/src/main/java/cm/aptoide/pt/social/view/viewholder/SocialMediaViewHolder.java

```diff
@@ -309,4 +310,48 @@ private void showNumberOfLikes(long likesNumber) {
         .toLowerCase());
     numberLikesOneLike.setVisibility(View.INVISIBLE);
   }
+
+  private void setupOverflowMenu(SocialMedia post, int position) {
+    overflowMenu.setOnClickListener(view -> {
+      PostPopupMenuBuilder postPopupMenuBuilder = new PostPopupMenuBuilder();
+      postPopupMenuBuilder.prepMenu(itemView.getContext(), overflowMenu)
+          .addReportAbuse(menuItem -> {
+            cardTouchEventPublishSubject.onNext(
+                new CardTouchEvent(post, position, CardTouchEvent.Type.REPORT_ABUSE));
+            return false;
+          });
+      accountManager.accountStatus()
```

Comment:
AptoideAccountManager to to model layer. To make it simple you can query account state every time Posts are requested. Ideally the Posts that require account status would change every time the account changes.

Reply:
fixed